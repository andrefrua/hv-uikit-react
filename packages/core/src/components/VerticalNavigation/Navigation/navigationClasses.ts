import { getClasses } from "@core/utils";

export interface HvVerticalNavigationTreeClasses {
  root?: string;
  list?: string;
  listItem?: string;
  collapsed?: string;
  popup?: string;
  navigationPopup?: string;
}

const classKeys: (keyof HvVerticalNavigationTreeClasses)[] = [
  "root",
  "list",
  "listItem",
  "collapsed",
  "popup",
  "navigationPopup",
];

const verticalNavigationTreeClasses = getClasses(
  classKeys,
  "HvVerticalNavigationTree"
);

export default verticalNavigationTreeClasses;
