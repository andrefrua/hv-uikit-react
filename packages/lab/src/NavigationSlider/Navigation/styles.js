const styles = (theme) => ({
  listItemRoot: {
    display: "flex",
    alignItems: "center",
    borderLeft: `2px solid transparent`,
    minHeight: "52px",
    marginBottom: "8px",
    "& > button": {
      marginLeft: "auto",
    },
  },
  listItemSelected: {
    background: theme.hv.palette.atmosphere.atmo3,
    borderLeft: `2px solid ${theme.hv.palette.accent.acce3}`,
    "& *": {
      background: theme.hv.palette.atmosphere.atmo3,
    },
  },
});

export default styles;
