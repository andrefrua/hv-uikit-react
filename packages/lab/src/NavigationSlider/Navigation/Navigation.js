import React from "react";
import { withStyles } from "@mui/styles";
import PropTypes from "prop-types";

import { HvListContainer, HvListItem, HvButton } from "@hitachivantara/uikit-react-core";
import { DropRightXS } from "@hitachivantara/uikit-react-icons";

import styles from "./styles";

const Navigation = ({ id, classes, data, selected, onNavigateToTarget, onNavigateToChild }) => {
  return (
    <HvListContainer interactive id={id}>
      {data.map((item) => (
        <HvListItem
          key={item.id}
          classes={{
            root: classes.listItemRoot,
            selected: classes.listItemSelected,
          }}
          onClick={(event) => onNavigateToTarget(event, item)}
          selected={selected === item.id}
          startAdornment={item.icon}
          endAdornment={
            item.data && item.data.length > 0 ? (
              <HvButton icon onClick={(event) => onNavigateToChild(event, item)}>
                <DropRightXS />
              </HvButton>
            ) : (
              <div />
            )
          }
        >
          <span>{item.label}</span>
        </HvListItem>
      ))}
    </HvListContainer>
  );
};

Navigation.propTypes = {
  /**
   * Id to be applied to the root node of the panel.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * styles object applied to the root
     */
    listItemRoot: PropTypes.string,
    listItemSelected: PropTypes.string,
  }).isRequired,
  /**
   * An array containing the data for each menu item.
   *
   * id - the id to be applied to the root element.
   * label - the label to be rendered on the menu item.
   * icon - the icon react element
   * data - sub-menu items
   * href - the url used for navigation.
   * target - the behavior when opening an url.
   */
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
      icon: PropTypes.node,
      data: PropTypes.array,
      href: PropTypes.string,
      target: PropTypes.string,
    })
  ).isRequired,
  /**
   * The selected item id.
   */
  selected: PropTypes.string,
  /**
   * Triggered when the item is clicked.
   */
  onNavigateToTarget: PropTypes.func,
  /**
   * Triggered when the navigate to child button is clicked.
   */
  onNavigateToChild: PropTypes.func,
};

export default withStyles(styles, { name: "Navigation" })(Navigation);
