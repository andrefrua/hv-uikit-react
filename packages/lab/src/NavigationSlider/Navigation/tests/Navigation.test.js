import React from "react";

import { create } from "react-test-renderer";

import { HvProvider, HvListContainer, HvListItem } from "@hitachivantara/uikit-react-core";

import Navigation from "..";

describe("<Navigation />", () => {
  let renderer;
  let instance;

  beforeAll(() => {
    const navigationData = [
      {
        id: "1",
        label: "Menu1",
        parent: undefined,
        icon: undefined,
        path: "/path1",
        data: [],
      },
    ];

    renderer = create(
      <HvProvider>
        <Navigation
          data={navigationData}
          selected="1"
          onNavigateToTarget={jest.fn}
          onNavigateToChild={jest.fn}
        />
      </HvProvider>
    );

    instance = renderer.root;
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it("should match the snapshot", () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("should be defined", () => {
    expect(renderer).toBeDefined();
  });

  it("should have a `HvListContainer` component present", async () => {
    const listContainers = await instance.findAllByType(HvListContainer);

    expect(listContainers.length).toBe(1);
  });

  it("should have a `HvListItem` component present", async () => {
    const listItems = await instance.findAllByType(HvListItem);

    expect(listItems.length).toBe(1);
  });
});

describe("<Navigation /> with different number of navigation items", () => {
  let renderer;
  let instance;
  beforeAll(() => {
    const navigationData = Array.from({ length: 10 }, (_, index) => {
      return {
        id: `${index + 1}`,
        label: `Menu${index}`,
        parent: undefined,
        icon: undefined,
        path: `/path${index}`,
        data: [],
      };
    });

    renderer = create(
      <HvProvider>
        <Navigation
          data={navigationData}
          selected="1"
          onNavigateToTarget={jest.fn}
          onNavigateToChild={jest.fn}
        />
      </HvProvider>
    );

    instance = renderer.root;
  });

  it("should have 10 HvListItem when there is 10 item in the Navigation data array", async () => {
    const listItems = await instance.findAllByType(HvListItem);

    expect(listItems.length).toBe(10);
  });
});
