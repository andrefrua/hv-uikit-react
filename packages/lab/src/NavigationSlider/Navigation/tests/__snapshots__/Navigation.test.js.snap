// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should match the snapshot 1`] = `
<ul
  className="HvListContainer-root"
>
  <li
    className="HvListItem-root Navigation-listItemRoot-9 HvListItem-gutters HvListItem-interactive HvListItem-selected Navigation-listItemSelected-10 HvListItem-withEndAdornment HvFocus-root HvFocus-selected HvFocus-focusDisabled"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    selected={true}
  >
    <span>
      Menu1
    </span>
    <div
      className="HvListItem-endAdornment"
    />
  </li>
</ul>
`;
